File: 06a/main.py
AFTER_0_VAL = 6
NEW_VAL = 8
DAYS = 80

with open('input.txt', 'rt') as f:
    fish = [int(x) for x in f.readline().split(',')]

for d in range(DAYS):
    new_fish = 0 
    for i, f in enumerate(fish):
        if f == 0:
            fish[i] = AFTER_0_VAL
            new_fish += 1
        else:
            fish[i] -= 1
    new_fish = [NEW_VAL for _ in range(new_fish)]
    fish.extend(new_fish)

result = len(fish)
print(result)

File: 06a/ruby/main.rb
FILENAME = "input.txt"
f = File.open(FILENAME)
fish = f.read.split(',').map(&:to_i)

DAYS = 80

$new_fish = 0

def tick(timer)
  if timer == 0
    $new_fish += 1
    6
  else
    timer - 1
  end
end

DAYS.times do
  $new_fish = 0
  fish = fish.map{ |x| tick(x) }
  fish += Array.new($new_fish, 8)
end

p fish.size
